edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/DeviceIdentityProvider.cs

```diff
@@ -48,6 +51,7 @@ public async Task<IDeviceIdentity> GetAsync(string clientId, string username, st
                 Preconditions.CheckNonWhiteSpace(clientId, nameof(clientId));
 
                 (string deviceId, string moduleId, string deviceClientType, Option<string> modelId) = ParseUserName(username);
+                modelId.ForEach(async m => await this.modelIdStore.SetModelId(deviceId, m));
```

Comment:
Interesting bug I found: If I put this line below any of the credentials stuff below (aka where productinfo setting is happening currently - line 94), I saw an exception being thrown, and then we weren't able to connect with the modelId. 
Putting it up here fixes the issue

Reply:
That is where we should do it. What was the exception being thrown?

Reply:
Microsoft.Azure.Devices.Edge.Hub.CloudProxy.DeviceScopeApiException: Message: Error getting device scope result from IoTHub, HttpStatusCode: Unauthorized, Content: {"Message":"ErrorCode:IotHubUnauthorizedAccess;Unauthorized","ExceptionMessage":"Tracking ID:[redacted]"}


I have made sure to set the parent of the device to be edgeHub, so I'm not sure why I'm getting this